api-inference-community/docker_images/timm/app/pipelines/image_classification.py

```diff
@@ -4,16 +4,26 @@
 import torch
 from app.pipelines import Pipeline
 from PIL import Image
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+from timm.models.hub import load_model_config_from_hf
 
 
 class ImageClassificationPipeline(Pipeline):
     def __init__(self, model_id: str):
-        if "resnet50d" in model_id:
-            self.model = timm.create_model("resnet50d", pretrained=True).eval()
-        elif "dpn92" in model_id:
-            self.model = timm.create_model("dpn92", pretrained=True).eval()
-        else:
-            raise EnvironmentError("Cannot detect the correct arch for this model")
+
+        self.hf_cfg, self.arch = load_model_config_from_hf(model_id)
+        self.model = timm.create_model(f"hf_hub:{model_id}", pretrained=True)
+        self.transform = create_transform(
+            **resolve_data_config(self.hf_cfg, model=self.model)
+        )
+        self.model.eval()
+
+        self.top_k = min(self.model.num_classes, 5)
+
+        self.labels = self.hf_cfg.get("labels", None)
+        if self.labels is None:
+            self.labels = [f"LABEL_{i}" for i in range(self.model.num_classes)]
```

Comment:
Got rid of the `IMAGENET_LABELS` list at the end of this file. If a user doesn't provide labels, we'll just use `LABEL_<idx>`. We'll update the official examples to include labels so they look nice in the inference API/widget.